src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1040,6 +1040,19 @@ internal void OnAdditionalDocumentTextUpdatedOnDisk(DocumentId documentId)
             this.OnAdditionalDocumentTextLoaderChanged(documentId, vsDoc.Loader);
         }
 
+        internal void OnIsComplete(ProjectId projectId, bool succeeded)
+        {
+            // this can't be null
+            var state = CurrentSolution.GetProjectState(projectId);
```

Comment:
sure, since it is in ProjectInfo. probably make sense to push it down to core workspace. will do.